osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/security/providers/HsmEncryptionProvider.java

```diff
@@ -45,63 +45,77 @@
 
     private final KeyStore keyStore;
 
-    public HsmEncryptionProvider(File keyStoreFile) {
+    public HsmEncryptionProvider(final File keyStoreFile) throws EncrypterException {
         try {
             super.setKeyFile(keyStoreFile);
             this.keyStore = KeyStore.getInstance(TYPE, PROVIDER);
-            FileInputStream fIn = new FileInputStream(keyStoreFile);
+            final FileInputStream fIn = new FileInputStream(keyStoreFile);
             this.keyStore.load(fIn, null);
-        } catch (CertificateException | NoSuchAlgorithmException | NoSuchProviderException | IOException | KeyStoreException e) {
+        } catch (final CertificateException | NoSuchAlgorithmException | NoSuchProviderException | IOException | KeyStoreException e) {
             throw new EncrypterException("Could not read keystore", e);
         }
     }
 
-    public Secret decrypt(EncryptedSecret secret, String keyReference) {
+    @Override
+    public byte[] decrypt(final EncryptedSecret secret, final String keyReference) throws EncrypterException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.opensmartgridplatform.shared.exceptionhandling.EncrypterException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)